projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-plugins-endpoint.php

```diff
@@ -144,7 +144,8 @@ protected function format_plugin( $plugin_file, $plugin_data ) {
 			$plugin['action_links'] = $action_link;
 		}
 
-		$autoupdate           = in_array( $plugin_file, (array) get_site_option( 'auto_update_plugins', array() ), true );
+		$plugin['plugin']     = $plugin_file;
+		$autoupdate           = ( new WP_Automatic_Updater() )->should_update( 'plugin', (object) $plugin, WP_PLUGIN_DIR );
```

Comment:
Would it be worth checking if the class exists before we use it, just in case it is not included in some environments?

Something like this:
https://github.com/Automattic/jetpack/blob/3753cb0c429b9ad9c9bd42fba05fbb925f4d310e/projects/packages/sync/src/class-functions.php#L270-L273

Reply:
Great idea, thanks! https://github.com/Automattic/jetpack/pull/18707/commits/7bc41ce18d887f51d46449887e3914417d3d1719